npc/re/other/achievement_treasures.txt

```diff
@@ -0,0 +1,342 @@
+//================= Hercules Script =======================================
+//=       _   _                     _
+//=      | | | |                   | |
+//=      | |_| | ___ _ __ ___ _   _| | ___  ___
+//=      |  _  |/ _ \ '__/ __| | | | |/ _ \/ __|
+//=      | | | |  __/ | | (__| |_| | |  __/\__ \
+//=      \_| |_/\___|_|  \___|\__,_|_|\___||___/
+//================= License ===============================================
+//= This file is part of Hercules.
+//= http://herc.ws - http://github.com/HerculesWS/Hercules
+//=
+//= Copyright (C) 2020 Hercules Dev Team
+//=
+//= Hercules is free software: you can redistribute it and/or modify
+//= it under the terms of the GNU General Public License as published by
+//= the Free Software Foundation, either version 3 of the License, or
+//= (at your option) any later version.
+//=
+//= This program is distributed in the hope that it will be useful,
+//= but WITHOUT ANY WARRANTY; without even the implied warranty of
+//= MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//= GNU General Public License for more details.
+//=
+//= You should have received a copy of the GNU General Public License
+//= along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//=========================================================================
+// Exploration Achievements
+//=========================================================================
+
+-	script	achievement_tr	FAKE_NPC,{
+	.@id = atoi(substr(strnpcinfo(NPC_NAME_HIDDEN), 2, getstrlen(strnpcinfo(NPC_NAME_HIDDEN)) - 1));
+	if (!achievement_iscompleted(.@id))
+		achievement_progress(.@id, 1, 1, true);
+	end;
+OnInit:
+	cloakonnpc(strnpcinfo(NPC_NAME));
+	end;
+}
+
+-	script	achievement_wp	FAKE_NPC,{
```

Comment:
why need 2 npc? cant merge the `OnTouch` into the `achievement_tr` ?  2 npc to maintain for each achievement npc if players ever decide to add a new one? and extra npcs mean less npc slot to use for other custom npc per map (although the limit around 512 per maps)

default npc with hidden sprite , then `clockoffnpc + changeclass` would do the trick.
i believe it would still trigger the event if player get near...

Reply:
That's because this is a conversion from the official script and I want to keep the same set of npcs to be able to reference it with that, in addition with this format the trigger area can be different from the chest area.